File: src/dumb_client.cpp
/* Copyright (c) 2015. The SimGrid Team. All rights reserved.              */

/* This program is free software; you can redistribute it and/or modify it
 * under the terms of the GNU Affero Licence (see in file LICENCE).        */

#include <stdio.h>
#include <rsg/actor.hpp>

int main(int argc, char **argv) {
	simgrid::rsg::Actor &self = simgrid::rsg::Actor::self();

	self.sleep(42);
	self.execute(15.*1000*1000*1000);
	self.send("toto","message from client");
	char * msg = self.recv("toto");
	fprintf(stderr, "Client: Received message: '%s'\n",msg);

	self.quit();
}

File: java/dumbClient.java
import org.simgrid.rsg.Actor;
import org.simgrid.rsg.Mailbox;

public class dumbClient {
	public static void main(String[] args) {

		// Start a bunch of threads (stress test)
		for (int i = 0; i<100; i++) {
			Thread t = new Thread() {
				@Override
				public void start() {
					Actor self = Actor.self();

					self.sleep(42);
					self.execute(8095000000.);
				}
			};
			t.run();
		}

		// A regular client doing some communications
		Actor self = Actor.self();
		Mailbox mbox = Mailbox.byName("toto");
		
		self.sleep(42);
		self.execute(8095000000.); // That's the power of my host on the used platform

		self.send(mbox,"message from client");
		String msg = self.recv(mbox);
		System.err.println("jClient: Received message: "+msg);
		
		self.quit();
	}
}
